lib/matplotlib/backend_bases.py

```diff
@@ -2040,12 +2044,45 @@ def draw_idle(self, *args, **kwargs):
             with self._idle_draw_cntx():
                 self.draw(*args, **kwargs)
 
+    @property
+    def pixel_ratio(self):
+        """
+        The ratio of logical to physical pixels used for the canvas.
+
+        Subclasses that support High DPI screens can set this property to
+        indicate that said ratio is different. The canvas itself will be
+        created at the physical size, while the display and any events will use
+        the logical size (and such implementations should ensure that events do
+        so.)
+
+        By default, this is 1, meaning physical and logical pixels are the same
+        size.
+        """
+        return self._pixel_ratio
+
+    @pixel_ratio.setter
+    def pixel_ratio(self, ratio):
```

Comment:
Debating whether this should be `set_pixel_ratio`, or even `_set_pixel_ratio` for only subclasses to call (and leave the property to be read-only.)

Reply:
IIUC third party users should never call this, only third-party backends that wrap other GUI toolkits and support HiDPI?  In which case I guess having just a private setter for now seems enough, and if the devs of such a backend come asking to make it public we can always revisit...

Reply:
I agree with @anntzer .  Safer to start private and make public if needed.  I also worry about this giving the impression that this setting will propagate farther back through the system that it does.